apps/web/src/store/collateral/hooks.ts

```diff
@@ -84,42 +123,146 @@ export function useCollateralAvailableAmount() {
   }, [ICXAmount]);
 }
 
-export function useCollateralAvailableAmountinSICX() {
-  const sicxAddress = bnJs.sICX.address;
-  const balances = useICONWalletBalances();
-  const sICXAmountCA = balances[sicxAddress];
-  const sICXAmount = toBigNumber(sICXAmountCA);
+export function useCollateralAmounts(xChainId?: XChainId): { [key in string]: BigNumber } {
+  const collateralXChain = useCollateralXChain();
+  return useSelector((state: AppState) => state.collateral.depositedAmounts[xChainId || collateralXChain] || {});
+}
+
+export function useAllCollateralData(): UseQueryResult<XCollaterals[]> {
+  const { data: totalCollateralData } = useTotalCollateralData();
 
-  return sICXAmount;
+  return useQuery({
```

Comment:
why did you use `useQuery`?

Reply:
https://github.com/balancednetwork/balanced-network-interface/issues/1388

Reply:
moved to https://github.com/balancednetwork/balanced-network-interface/issues/1388

Reply:
@0xmilktea to have to result cached